manifests/init.pp

```diff
@@ -75,8 +80,11 @@
   Optional[String] $agent_old_password = undef,
 ) {
 
-  if $use_ssl and ! $ssl_ca_source {
-    fail('sensu: ssl_ca_source must be defined when use_ssl is true')
+  if $use_ssl and $ssl_ca_source and $ssl_ca_content {
+    fail('sensu::backend: Do not define both ssl_ca_source and ssl_ca_content_content')
+  }
+  if $use_ssl and ! ($ssl_ca_source or $ssl_ca_content) {
+    $ssl_ca_source = $facts['puppet_localcacert']
```

Comment:
Can't reassign a variable in Puppet. This would have to be named `$_ssl_ca_source` and that value used.

Simpler approach:

```
if $ssl_ca_content {
  $_ssl_ca_source = undef
} else {
  $_ssl_ca_source = pick($ssl_ca_source, $facts['puppet_localcacert'])
}
```

Will need to have two validation conditions

```
if $use_ssl {
  if $ssl_ca_source and $ssl_ca_content {
    ... error they can't be defined at same time ...
  }
  if ! ($ssl_ca_source and $ssl_ca_content) {
    ... ssl_ca_source or ssl_ca_content must be defined when use_ssl is true ...
  }
}
```